### PR TITLE
Fix deprecation warnings

### PR DESCRIPTION
**Pull request by joschi**
This PR fixes a variety of deprecation warnings.

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/configuration/ExposedConfiguration.java [hunk 1]
```diff
@@ -64,24 +63,6 @@
     @JsonProperty("allow_leading_wildcard_searches")
     public abstract boolean allowLeadingWildcardSearches();
 
-    @JsonProperty("rotation_strategy")
```
Comment (bernd) [inline graylog2-server/src/main/java/org/graylog2/configuration/ExposedConfiguration.java]: Please update the UPGRADING file to document the changed `/system/configuration` response.

**Comment by bernd** [comment]
And please rebase against master, thanks! :smiley:

File: graylog2-server/src/main/java/org/graylog2/configuration/ExposedConfiguration.java (hunk 1 above)
Comment (joschi) [inline graylog2-server/src/main/java/org/graylog2/configuration/ExposedConfiguration.java]: Done.

**Comment by bernd** [comment]
LGTM :+1:

**Pull request merged**